guidelines/monai-application-package.md

```diff
@@ -1,498 +1,588 @@
+- Developers expect the local execution of their applications to behave identically to the execution of the containerized version.
 
 - Developers expect local execution of their containerized applications to behave identically to execution in deployment.
 
 - Developers and operations engineers want the application packages to be self-describing.
 
-- Application packages might not be developed using the MONAI Application SDK.
+- Applications might not be developed using the Holoscan SDK or an SDK based on it, e.g., MONAI Deploy App SDK.
+
+- MONAI Application Package may not be created using the Holoscan SDK nor the MONAI Deploy App SDK.
 
-- Pre-existing, containerized applications will need to be "converted" into MONAI Application Packages.
+- Pre-existing, containerized applications must be "converted" into MONAI Application Packages.
 
-- Application packages will be deployed in a variety of environments.
+- The scalability of a multi-fragment application based on Holoscan SDK is outside the scope of this document.
+
+- A MONAI Application Package may contain a classical application (non-fragment based), a single-fragment application, or a multi-fragment application.
+
+- Application packages are expected to be deployed in a variety of environments.
 
   > [Note]
   > See [MONAI Operating Environments](monai-operating-environments.md) for additional information about supported environments.
 
+## Requirements
 
-##  Requirements
+The following are requirements that need to be met by the MAP specification to be considered complete and approved.
 
-The following are requirements which need to be met by the MAP specification to be considered complete and approved.
+### Single Artifact
 
+- A MAP SHALL comprise a single container meeting the minimum requirements this document sets forth.
+- A MAP SHALL be a containerized application to maximize the portability of its application.
 
-### Contains an Application
+### Self-Describing
 
-A MAP SHALL contain an executable application which supports the primary function of the MAP, and provide sufficient information to execute the application as intended.
+- A MAP MUST be self-describing and provide a mechanism for extracting its description.
+  - A MAP SHALL provide a method to print the metadata files to the console.
+  - A MAP SHALL provide a method to copy the metadata files to the user-specified directory.
+- The method of description SHALL be in a machine-readable and writable format.
+- The method of description SHALL be in a human-readable format.
+- The method of description SHOULD be a human writable format.
+- The method of description SHALL be declarative and immutable.
+- The method of description SHALL provide the following information about the MAP:
+  - Execution requirements such as dependencies and restrictions.
+  - Resource requirements include CPU cores, system memory, and GPU availability.
```

Comment:
"GPU availability"? Do you mean GPU suitability, requirements or something else?

Reply:
[Resource requirements include CPU cores, system memory, shared memory, GPU, and GPU memory